bridgesync/processor.go

```diff
@@ -505,7 +523,71 @@ func (p *processor) GetBridges(
 	return bridges, nil
 }
 
-func (p *processor) GetClaims(ctx context.Context, fromBlock, toBlock uint64) ([]Claim, error) {
+func (p *processor) GetClaims(ctx context.Context, fromBlock, toBlock uint64, compacted bool) ([]Claim, error) {
+	if !compacted {
+		return p.getClaims(ctx, fromBlock, toBlock)
+	}
+
+	// Create a context with database timeout
+	dbCtx, cancel := p.withDatabaseTimeout(ctx)
+	defer cancel()
+
+	// SQL query with compaction logic:
+	// - Gets the oldest claim for each global_index (preserves all metadata)
+	// - Updates only the proof fields from the newest claim with same global_index
+	query := fmt.Sprintf(`
+		WITH ranked_claims AS (
+			SELECT 
+				*,
+				ROW_NUMBER() OVER (PARTITION BY global_index ORDER BY block_num ASC, block_pos ASC) AS rn_oldest,
+				ROW_NUMBER() OVER (PARTITION BY global_index ORDER BY block_num DESC, block_pos DESC) AS rn_newest
+			FROM claim
+			WHERE block_num >= $1 AND block_num <= $2
+		),
+		oldest_with_newest_proofs AS (
+			SELECT 
+			%s
+			FROM ranked_claims o
+			JOIN ranked_claims n ON o.global_index = n.global_index AND n.rn_newest = 1
+			WHERE o.rn_oldest = 1
+		)
+		SELECT * FROM oldest_with_newest_proofs
+		ORDER BY block_num ASC, block_pos ASC;
+	`, compactedClaimsSelectSQL)
+
+	rows, err := p.db.QueryContext(dbCtx, query, fromBlock, toBlock)
+	if err != nil {
+		if errors.Is(err, sql.ErrNoRows) {
+			p.log.Debugf("no claims were found for block range [%d..%d]", fromBlock, toBlock)
+			return []Claim{}, nil
+		}
+		p.log.Errorf("GetClaims query failed for block range [%d..%d]: %v", fromBlock, toBlock, err)
+		return nil, err
+	}
+
+	defer func() {
+		if cerr := rows.Close(); cerr != nil {
+			p.log.Errorf("error closing rows: %v", cerr)
+		}
+	}()
+
+	claimPtrs := []*Claim{}
+	if err = meddler.ScanAll(rows, &claimPtrs); err != nil {
+		p.log.Errorf("GetClaims: meddler.ScanAll failed for block range [%d..%d]: %v", fromBlock, toBlock, err)
+		return nil, err
+	}
+
+	claimsIface := db.SlicePtrsToSlice(claimPtrs)
+	claims, ok := claimsIface.([]Claim)
+	if !ok {
+		p.log.Errorf("GetClaims: failed to convert from []*Claim to []Claim for block range [%d..%d]", fromBlock, toBlock)
+		return nil, errFailToConvertClaims
+	}
+
+	return claims, nil
+}
+
+func (p *processor) getClaims(ctx context.Context, fromBlock, toBlock uint64) ([]Claim, error) {
```

Comment:
We can probably change getClaims function, so that it receives query string so that this [part](https://github.com/agglayer/aggkit/blob/eabc00da5a634a9b6750e658674134ce92133e08/bridgesync/processor.go#L558-L585) reuses this function (or something similar) to avoid code repetitions. The idea is that both compacted and non compacted versions are relying on same `getClaimsInternal` function. WDYT?